tests/test_web_websocket.py

```diff
@@ -412,3 +412,33 @@ async def test_no_transfer_encoding_header(make_request: Any, mocker: Any) -> No
     await ws._start(req)
 
     assert "Transfer-Encoding" not in ws.headers
+
+
+async def test_get_extra_info(make_request: Any, mocker: Any) -> None:
+    valid_key = "test"
+    valid_value = "existent"
+    default_value = "default"
+
+    def get_extra_info(name: str, default: Any = None):
+        return {valid_key: valid_value}.get(name, default)
+
+    transp = mock.Mock()
+    transp.get_extra_info.side_effect = get_extra_info
+
+    req = make_request("GET", "/")
+    ws = WebSocketResponse()
+
+    await ws.prepare(req)
+    ws._writer.transport = transp
+
+    ws_extra_info = ws.get_extra_info(valid_key, default_value)
+    transp_extra_info = ws._writer.transport.get_extra_info(valid_key, default_value)
+    assert ws_extra_info == transp_extra_info
+
+    ws._writer.transport = None
+    extra_info = ws.get_extra_info(valid_key, default_value)
+    assert extra_info == default_value
+
+    ws._writer = None
+    extra_info = ws.get_extra_info(valid_key, default_value)
+    assert extra_info == default_value
```

Comment:
Looks like you included 3 separate tests into the same test function. This is not a good idea as they wouldn't show up in the test report. Also, if one assert fails, the following ones won't be checked, meaning that it's impossible to know if other cases pass, from the test report output.

To fix this, use parametrize. You can also use param ids to label individual tests for human readability.

Reply:
fixed